demos/faults.h

```diff
@@ -14,13 +14,13 @@
 
 #include <functional>
 
-// Run `inner` with a signal handler installed to catch failure signals,
-// currently defined as `SIGSEGV`. If such a signal is raised, the execution
-// of `inner` is aborted.
+// Run `inner` with a signal handler installed for `fault_signum`, which is
+// probably one of: SIGSEGV, SIGBUS, SIGFPE, SIGILL, or SIGTRAP. If that signal
```

Comment:
Can we document the rationale for choice? Like I think at least with SIGSEGV, SIGBUS, and SIGILL, the main benefit here is that we know that valid code won't raise those signals, so the only source will be us. (Less sure what motivates the inclusion of SIGFPE. Maybe because we don't use floats, we get the same benefit?)

Reply:
Added to the comment.